kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -183,18 +188,78 @@ object Abort:
                     handle = [C] => (input, _) => input,
                     recover =
                         case ct(fail) if ct <:< SafeClassTag[Throwable] =>
-                            Result.fail(fail)
+                            continue(Result.Failure(fail))
                         case fail =>
-                            Result.panic(fail)
+                            continue(Result.Panic(fail)),
+                    done = continue(_)
                 )
             }
+    end RunWithOps
+
+    /** Runs an Abort effect. This operation handles the Abort effect, converting it into a Result type.
+      */
+    inline def runWith[E]: RunWithOps[E] = RunWithOps(())
+
+    final class RunOps[E >: Nothing](dummy: Unit) extends AnyVal:
+        /** Runs an Abort effect, converting it to a Result.
+          *
+          * @param v
+          *   The computation to run
+          * @tparam A
+          *   The success type of the computation
+          * @tparam S
+          *   The effect type of the computation
+          * @tparam ER
+          *   Any remaining Abort effects after running this one
+          * @return
+          *   A Result containing either the success value or the failure value, wrapped in the remaining effects
+          */
+        def apply[A: Flat, S, ER](v: => A < (Abort[E | ER] & S))(
+            using
+            frame: Frame,
+            ct: SafeClassTag[E],
+            reduce: Reducible[Abort[ER]]
+        ): Result[E, A] < (S & reduce.SReduced) =
+            runWith[E](v)(identity)
         end apply
     end RunOps
 
     /** Runs an Abort effect. This operation handles the Abort effect, converting it into a Result type.
       */
     inline def run[E]: RunOps[E] = RunOps(())
 
+    final class RunPartialOps[E >: Nothing](dummy: Unit) extends AnyVal:
+        /** Runs an Abort effect, converting it to a partial Result.
+          *
+          * @param v
+          *   The computation to run
+          * @tparam A
+          *   The success type of the computation
+          * @tparam S
+          *   The effect type of the computation
+          * @tparam ER
+          *   Any remaining Abort effects after running this one
+          * @return
+          *   A Result containing either the success value or the failure value, wrapped in the remaining effects
+          */
+        def apply[A: Flat, S, ER](v: => A < (Abort[E | ER] & S))(
+            using
+            frame: Frame,
+            ct: SafeClassTag[E],
+            f2: Flat[Result.Partial[E, A]],
+            reduce: Reducible[Abort[ER]]
+        ): Result.Partial[E, A] < (S & reduce.SReduced) =
+            Abort.runWith[E](v):
+                case Panic(thr)                      => throw thr
```

Comment:
This doesn't seem correct. The returned computation should have a pending `Abort[Nothing]` given that panics aren't handled. I think we can remove the `Reducible` evidence and return `Result.Partial[E, A] < (S & Abort[ER])`, which will infer to `Abort[Nothing]` if it's the only failure pending.

Reply:
I've added some additional `...OrThrow` methods to handle completely and throwing in the case of `Panic`.